### PR TITLE
chore(cli): correctly set up `cdk watch` in init templates

### PR DESCRIPTION
**Pull request by comcalvi**
This enables `watch` to work out-of-the-box by adding the `watch` key to all init template `cdk.json` files. By default `watch` will look only at source files; project configuration files will not be included by `watch`. The `build` key is not needed for any language, as all of the `app` commands will build the code before it is executed. 

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by skinny85** [comment]
@eladb do you mind taking a look as well? Thanks!

File: packages/aws-cdk/lib/init-templates/v2/sample-app/typescript/cdk.template.json
Comment (skinny85) [inline packages/aws-cdk/lib/init-templates/v2/sample-app/typescript/cdk.template.json:5]: JS files should also be excluded I think?

File: packages/aws-cdk/lib/init-templates/v2/sample-app/typescript/cdk.template.json
Comment (skinny85) [inline packages/aws-cdk/lib/init-templates/v2/sample-app/typescript/cdk.template.json:4]: Too broad (also includes test files).

File: packages/aws-cdk/lib/init-templates/v2/sample-app/python/cdk.template.json
Comment (skinny85) [inline packages/aws-cdk/lib/init-templates/v2/sample-app/python/cdk.template.json:4]: Too broad (also includes test files).

File: packages/aws-cdk/lib/init-templates/v2/sample-app/javascript/cdk.template.json
Comment (skinny85) [inline packages/aws-cdk/lib/init-templates/v2/sample-app/javascript/cdk.template.json:4]: Too broad (also includes test files).

File: packages/aws-cdk/lib/init-templates/v2/sample-app/java/cdk.json
Comment (skinny85) [inline packages/aws-cdk/lib/init-templates/v2/sample-app/java/cdk.json:4]: Too broad (also includes test files).

File: packages/aws-cdk/lib/init-templates/v2/sample-app/java/cdk.json
Comment (skinny85) [inline packages/aws-cdk/lib/init-templates/v2/sample-app/java/cdk.json:5]: Pretty sure `target` should be excluded...?

File: packages/aws-cdk/lib/init-templates/v2/sample-app/go/cdk.template.json
Comment (skinny85) [inline packages/aws-cdk/lib/init-templates/v2/sample-app/go/cdk.template.json:4]: Too broad (also includes test files).

File: packages/aws-cdk/lib/init-templates/v2/sample-app/fsharp/cdk.template.json
Comment (skinny85) [inline packages/aws-cdk/lib/init-templates/v2/sample-app/fsharp/cdk.template.json:4]: Too broad (also includes test files).

File: packages/aws-cdk/lib/init-templates/v2/sample-app/csharp/cdk.template.json
Comment (skinny85) [inline packages/aws-cdk/lib/init-templates/v2/sample-app/csharp/cdk.template.json:4]: Too broad (also includes test files).

File: packages/aws-cdk/lib/init-templates/v2/app/typescript/cdk.template.json
Comment (skinny85) [inline packages/aws-cdk/lib/init-templates/v2/app/typescript/cdk.template.json:4]: Too broad (also includes test files).

Comment (skinny85) [inline packages/aws-cdk/lib/init-templates/v2/app/typescript/cdk.template.json:4]: Also, this is inconsistent with the other templates:

```suggestion
    "include": "**/*.ts",
```

Or change the other templates to use `"./**/sth"` instead of `"**/sth"`, but it should be consistent among them.

File: packages/aws-cdk/lib/init-templates/v2/app/python/cdk.template.json
Comment (skinny85) [inline packages/aws-cdk/lib/init-templates/v2/app/python/cdk.template.json:4]: Too broad (also includes test files).

File: packages/aws-cdk/lib/init-templates/v2/app/javascript/cdk.template.json
Comment (skinny85) [inline packages/aws-cdk/lib/init-templates/v2/app/javascript/cdk.template.json:4]: Too broad (also includes test files).

File: packages/aws-cdk/lib/init-templates/v2/app/java/cdk.json
Comment (skinny85) [inline packages/aws-cdk/lib/init-templates/v2/app/java/cdk.json:4]: Too broad (also includes test files).

File: packages/aws-cdk/lib/init-templates/v2/app/go/cdk.template.json
Comment (skinny85) [inline packages/aws-cdk/lib/init-templates/v2/app/go/cdk.template.json:4]: I actually don't know if the Go template includes tests by default, but worth checking.

File: packages/aws-cdk/lib/init-templates/v1/sample-app/typescript/cdk.template.json
Comment (skinny85) [inline packages/aws-cdk/lib/init-templates/v1/sample-app/typescript/cdk.template.json:4]: Too broad (also includes test files).

File: packages/aws-cdk/lib/init-templates/v1/sample-app/javascript/cdk.template.json
Comment (skinny85) [inline packages/aws-cdk/lib/init-templates/v1/sample-app/javascript/cdk.template.json:4]: Too broad (also includes test files).

File: packages/aws-cdk/lib/init-templates/v1/sample-app/java/cdk.json
Comment (skinny85) [inline packages/aws-cdk/lib/init-templates/v1/sample-app/java/cdk.json:4]: Too broad (also includes test files).

File: packages/aws-cdk/lib/init-templates/v1/sample-app/go/cdk.template.json
Comment (skinny85) [inline packages/aws-cdk/lib/init-templates/v1/sample-app/go/cdk.template.json:4]: Same comment for all of these and above as below, I don't want to repeat myself (for some reason, I went bottom to top on this review 😜).

**Review by skinny85 — CHANGES_REQUESTED** [review]
Generally, looks good! I think some of these do not correctly exclude tests, so that needs to be fixed.

I'm also thinking whether it wouldn't be better to be more wide with inclusions, and more specific with exclusions, than the opposite. The current approach seems to favor the latter, for example:

```json
  "watch": {
    "include": "./**/*.ts",
    "exclude": "./**/*.d.ts"
  }
```

I'm wondering whether this:

```json
  "watch": {
    "include": "**/*",
    "exclude": ["./**/*.d.ts", "./**/*.js"]
  }
```

Wouldn't be more "future proof"?

@eladb what are your thoughts on this?

Also, let's use arrays everywhere, even if we use only a single value, so that folks will now they are allowed multiple values here when they are editing these.

File: packages/aws-cdk/lib/init-templates/v1/app/python/cdk.template.json
Comment (eladb) [inline packages/aws-cdk/lib/init-templates/v1/app/python/cdk.template.json:5]: Why?

File: packages/aws-cdk/lib/init-templates/v1/app/typescript/cdk.template.json
Comment (eladb) [inline packages/aws-cdk/lib/init-templates/v1/app/typescript/cdk.template.json:4]: I'd include node_modules here for completeness (also in javascript). 

I know it's implicitly added but still

File: packages/aws-cdk/lib/init-templates/v1/sample-app/javascript/cdk.template.json
Comment (eladb) [reply packages/aws-cdk/lib/init-templates/v1/sample-app/javascript/cdk.template.json:4]: Why not exclude test/?

**Comment by eladb** [comment]
> Generally, looks good! I think some of these do not correctly exclude tests, so that needs to be fixed.
> 
> 
> 
> I'm also thinking whether it wouldn't be better to be more wide with inclusions, and more specific with exclusions, than the opposite. The current approach seems to favor the latter, for example:
> 
> 
> 
> ```json
> 
>   "watch": {
> 
>     "include": "./**/*.ts",
> 
>     "exclude": "./**/*.d.ts"
> 
>   }
> 
> ```
> 
> 
> 
> I'm wondering whether this:
> 
> 
> 
> ```json
> 
>   "watch": {
> 
>     "include": "**/*",
> 
>     "exclude": ["./**/*.d.ts", "./**/*.js"]
> 
>   }
> 
> ```
> 
> 
> 
> Wouldn't be more "future proof"?
> 
> 
> 
> @eladb what are your thoughts on this?
> 
> 
> 
> Also, let's use arrays everywhere, even if we use only a single value, so that folks will now they are allowed multiple values here when they are editing these.
> 
> 

+1 on excluding js files but why include non .ts files? Is that in order to monitor changes in assets?

**Comment by skinny85** [comment]
I think the `"include"` is worth it, even if it's redundant, mainly to make it obvious for developers how to edit those settings, without the need to check our documentation. Basically, the same reason why I think we should always use arrays in the init templates for `"include"` and `"exclude"`, even when the array has one element, and those keys accept single strings.

File: packages/aws-cdk/lib/init-templates/v1/app/javascript/cdk.template.json
Comment (skinny85) [inline packages/aws-cdk/lib/init-templates/v1/app/javascript/cdk.template.json:7]: Let's maybe add `yarn.lock` here too?

Same comment in the other JS/TS templates.

File: packages/aws-cdk/lib/init-templates/v1/app/javascript/cdk.template.json
Comment (skinny85) [inline packages/aws-cdk/lib/init-templates/v1/app/javascript/cdk.template.json:10]: I think Elad wanted to exclude `node_modules` explicitly too.

Same comment in the other JS/TS templates.

File: packages/aws-cdk/lib/init-templates/v1/app/python/cdk.template.json
Comment (skinny85) [inline packages/aws-cdk/lib/init-templates/v1/app/python/cdk.template.json:12]: This should be indented with 4 spaces, not 2.

```suggestion
            "**/__init__.py",
            "README.md",
            "cdk.json",
            "*.txt",
            "source.bat",
            "tests",
          "python/__pycache__"
```

File: packages/aws-cdk/lib/init-templates/v1/app/python/cdk.template.json
Comment (skinny85) [inline packages/aws-cdk/lib/init-templates/v1/app/python/cdk.template.json:9]: Seems too wide. `"requirements*.txt"`?

File: packages/aws-cdk/lib/init-templates/v1/app/typescript/cdk.template.json
Comment (skinny85) [inline packages/aws-cdk/lib/init-templates/v1/app/typescript/cdk.template.json:11]: Let's exclude `jest.config.js` only, like we do in the JS templates.

File: packages/aws-cdk/lib/init-templates/v1/sample-app/csharp/cdk.template.json
Comment (skinny85) [inline packages/aws-cdk/lib/init-templates/v1/sample-app/csharp/cdk.template.json:14]: Same comments here.

File: packages/aws-cdk/lib/init-templates/v1/sample-app/fsharp/cdk.template.json
Comment (skinny85) [inline packages/aws-cdk/lib/init-templates/v1/sample-app/fsharp/cdk.template.json:13]: Same comments here.

File: packages/aws-cdk/lib/init-templates/v1/sample-app/javascript/cdk.template.json
Comment (skinny85) [inline packages/aws-cdk/lib/init-templates/v1/sample-app/javascript/cdk.template.json:12]: Same comments here.

File: packages/aws-cdk/lib/init-templates/v1/sample-app/python/cdk.template.json
Comment (skinny85) [inline packages/aws-cdk/lib/init-templates/v1/sample-app/python/cdk.template.json:9]: Same comments here.

File: packages/aws-cdk/lib/init-templates/v1/sample-app/typescript/cdk.template.json
Comment (skinny85) [inline packages/aws-cdk/lib/init-templates/v1/sample-app/typescript/cdk.template.json:11]: Same comments here.

File: packages/aws-cdk/lib/init-templates/v2/app/csharp/cdk.template.json
Comment (skinny85) [inline packages/aws-cdk/lib/init-templates/v2/app/csharp/cdk.template.json:9]: Same comments here.

File: packages/aws-cdk/lib/init-templates/v2/app/fsharp/cdk.template.json
Comment (skinny85) [inline packages/aws-cdk/lib/init-templates/v2/app/fsharp/cdk.template.json:8]: Same comments here.

File: packages/aws-cdk/lib/init-templates/v2/app/javascript/cdk.template.json
Comment (skinny85) [inline packages/aws-cdk/lib/init-templates/v2/app/javascript/cdk.template.json:10]: Same comments here.

File: packages/aws-cdk/lib/init-templates/v2/app/python/cdk.template.json
Comment (skinny85) [inline packages/aws-cdk/lib/init-templates/v2/app/python/cdk.template.json:9]: Same comments here.

File: packages/aws-cdk/lib/init-templates/v2/app/typescript/cdk.template.json
Comment (skinny85) [inline packages/aws-cdk/lib/init-templates/v2/app/typescript/cdk.template.json:12]: Same comments here.

File: packages/aws-cdk/lib/init-templates/v2/sample-app/csharp/cdk.template.json
Comment (skinny85) [inline packages/aws-cdk/lib/init-templates/v2/sample-app/csharp/cdk.template.json:8]: Same comments here.

File: packages/aws-cdk/lib/init-templates/v2/sample-app/fsharp/cdk.template.json
Comment (skinny85) [inline packages/aws-cdk/lib/init-templates/v2/sample-app/fsharp/cdk.template.json:8]: Same comments here.

File: packages/aws-cdk/lib/init-templates/v2/sample-app/javascript/cdk.template.json
Comment (skinny85) [inline packages/aws-cdk/lib/init-templates/v2/sample-app/javascript/cdk.template.json:7]: Same comments here.

File: packages/aws-cdk/lib/init-templates/v2/sample-app/python/cdk.template.json
Comment (skinny85) [inline packages/aws-cdk/lib/init-templates/v2/sample-app/python/cdk.template.json:9]: Same comments here.

File: packages/aws-cdk/lib/init-templates/v2/sample-app/typescript/cdk.template.json
Comment (skinny85) [inline packages/aws-cdk/lib/init-templates/v2/sample-app/typescript/cdk.template.json:11]: Same comments here.

**Review by skinny85 — CHANGES_REQUESTED** [review]
We're getting there! A few more comments.

File: packages/aws-cdk/lib/init-templates/v1/app/csharp/cdk.template.json
Comment (skinny85) [inline packages/aws-cdk/lib/init-templates/v1/app/csharp/cdk.template.json:8]: This is better, but I have an even better idea 🙂:

```suggestion
      "src/*/obj",
```

What do you think about doing that instead @comcalvi?

Comment (comcalvi) [reply packages/aws-cdk/lib/init-templates/v1/app/csharp/cdk.template.json:8]: I don't see why not, but I also don't see why. What's the advantage @skinny85?

Comment (skinny85) [reply packages/aws-cdk/lib/init-templates/v1/app/csharp/cdk.template.json:8]: 1. It's shorter.
2. If you rename your app, you don't have to change these settings.

File: packages/aws-cdk/lib/init-templates/v1/sample-app/python/cdk.template.json
Comment (skinny85) [inline packages/aws-cdk/lib/init-templates/v1/sample-app/python/cdk.template.json:7]:
```suggestion
            "cdk*.json",
```

File: packages/aws-cdk/lib/init-templates/v1/app/typescript/cdk.template.json
Comment (skinny85) [inline packages/aws-cdk/lib/init-templates/v1/app/typescript/cdk.template.json:10]: I _think_ this is covered by the above JS file exclusion already, no?

Same comment in the other TS templates.

File: packages/aws-cdk/lib/init-templates/v1/app/csharp/cdk.template.json
Comment (skinny85) [inline packages/aws-cdk/lib/init-templates/v1/app/csharp/cdk.template.json:12]: Wouldn't it make more sense to just exclude all `.csproj` files? I can't imagine we would ever want to track those.

Same comment in the other C# and F# templates.

File: packages/aws-cdk/lib/init-templates/v2/app/python/cdk.template.json
Comment (skinny85) [inline packages/aws-cdk/lib/init-templates/v2/app/python/cdk.template.json:7]:
```suggestion
            "cdk*.json",
```

File: packages/aws-cdk/lib/init-templates/v1/app/typescript/cdk.template.json
Comment (skinny85) [inline packages/aws-cdk/lib/init-templates/v1/app/typescript/cdk.template.json:13]: This order seems inconsistent with the JS templates. I think the JS template order is the correct one we want to use here (first `"package*.json"`, then `"yarn.lock"`).

File: packages/aws-cdk/lib/init-templates/v2/sample-app/python/cdk.template.json
Comment (skinny85) [inline packages/aws-cdk/lib/init-templates/v2/sample-app/python/cdk.template.json:7]:
```suggestion
            "cdk*.json",
```

**Review by skinny85 — CHANGES_REQUESTED** [review]
Looks like the Python templates don't have the exclusions for `"cdk.context.json"` updated, but other than that, looks great! Added also a few minor optional comments.

**Review by skinny85 — APPROVED** [review]
Looks great to me, everything is top-notch - nice work @comcalvi!

@eladb I'm giving you a chance for a last view before we merge this in, let us know if you have any comments!

**Pull request merged**
